R/logging.R

```diff
@@ -0,0 +1,110 @@
+#' Get node log files from Azure Storage. By default, this operation will print the files on screen.
+#'
+#' @param cluster The cluster object
+#' @param nodeId Id of the node
+#' @param type The type of logs the user wants: "stdout" or "stderr"
+#' @param verbose Flag for printing log files onto console
+#'
+#' @param ... Further named parameters
+#' \itemize{
+#'  \item{"localPath"}: { Path to save file to }
+#'  \item{"overwrite"}: { Will only overwrite existing localPath }
+#'}
+#'
+#' @examples
+#' stdoutText <- getClusterLogs(cluster, "tvm-1170471534_1-20170829t072146z", type = "stdout", verbose = FALSE)
+#' getClusterLogs(cluster, "tvm-1170471534_2-20170829t072146z", type = "stderr", localPath = "abc.txt", overwrite = TRUE)
+#' @export
+getClusterLogs <-
+  function(cluster,
+           nodeId,
+           type = c("stdout", "stderr"),
+           verbose = TRUE,
+           overwrite = FALSE,
+           localPath = NULL) {
+    filePathString <- "startup/%s.txt"
```

Comment:
Don't love this... Also, as per my comment above, I really think we can just accept a file name.

To that point, we can rename this method to getClusterFile and it magically becomes more generic :)

Reply:
uhh i think i want to push back on the getClusterFile bit. 

I guess i could see that making sense as a separate thing, but getClusterLogs as an abstraction for the user seems important (maybe it is just a convenience function).

In my mind, getJobLogs is more important - so maybe the more generic getClusterFile works...